website/content/docs/concepts/architecture.md

```diff
@@ -213,3 +213,53 @@ The source for this diagram lives here:
 https://docs.google.com/drawings/d/10d1Q_VO0HFtOHftDV7kK6VbZMVI5EpEYHrD-LR7SczE
 -->
 <img src="/img/topology-submit-sequence-diagram.png" style="max-width:140%;!important;" alt="Topology Sequence Diagram"/>
+
+### Brief Description
+
+The following briefly describes how a topology is handled internally.
+
+* Client
+
+    When a topology is submitted using `heron submit` command, it first executes
+    the `main` of the topology and creates `.defn` file containing the topology's
+    logical plan. Then, it runs `com.twitter.heron.scheduler.SubmitterMain`, which
+    is responsible for invoking an uploader and a launcher for the topology.
+    The uploader uploads the topology package to the given location, while the
+    launcher registers the topology's logical plan and executor state with 
+    the state manager and invokes the main scheduler.
+
+* Shared Services
+
+    When the main scheduler (`com.twitter.heron.scheduler.SchedulerMain`) is invoked 
+    by the launcher, it fetches the submitted topology artifact from the
+    topology storage, initializes the state manager, and prepares the packed plan, which 
+    specifies how multiple instances should be packed into containers. Then, it starts
+    the specified scheduler, such as `com.twitter.heron.scheduler.local.LocalScheduler`,
+    which invokes `heron-executor` for each container.
+
+* Topologies
+
+    `heron-executor` process is run for each container and is responsible for 
+    executing the topology master or heron instances (bolt/spout) that are
+    assigned to the container. This instance distribution is determined
+    by the packed plan. Note that the topology master is always executed 
+    on the container 0. When `heron-executor` executes 
+    normal heron instances (i.e. except for container 0), it first prepares 
+    the stream manager and the metrics manager, before starting 
+    `com.twitter.heron.instance.HeronInstance` for each instance that is 
+    assigned to the container. 
+    
+    Heron Instance has two threads: Gateway thread and Slave thread. Gateway
```

Comment:
Gateway and Slave shouldn't be capitalized here. Also, this should be "the gateway thread and the slave thread."